tool/tctl/common/resource_command.go

```diff
@@ -594,78 +592,105 @@ func (rc *ResourceCommand) getCollection(client auth.ClientI) (c ResourceCollect
+		return &samlCollection{connectors}, nil
 	case services.KindOIDCConnector:
-		connectors, err := client.GetOIDCConnectors(ctx, rc.withSecrets)
+		connectors, err := getOIDCConnectors(ctx, client, rc.ref.Name, rc.withSecrets)
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
-		return &oidcCollection{connectors: connectors}, nil
+		return &oidcCollection{connectors}, nil
 	case services.KindGithubConnector:
-		connectors, err := client.GetGithubConnectors(ctx, rc.withSecrets)
+		connectors, err := getGithubConnectors(ctx, client, rc.ref.Name, rc.withSecrets)
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
-		return &githubCollection{connectors: connectors}, nil
+		return &githubCollection{connectors}, nil
 	case services.KindReverseTunnel:
-		tunnels, err := client.GetReverseTunnels()
-		if err != nil {
-			return nil, trace.Wrap(err)
+		if rc.ref.Name == "" {
+			tunnels, err := client.GetReverseTunnels()
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+			return &reverseTunnelCollection{tunnels: tunnels}, nil
 		}
-		return &reverseTunnelCollection{tunnels: tunnels}, nil
-	case services.KindCertAuthority:
-		var authorities []services.CertAuthority
-
-		userAuthorities, err := client.GetCertAuthorities(services.UserCA, rc.withSecrets)
+		tunnel, err := client.GetReverseTunnel(rc.ref.Name)
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
-		authorities = append(authorities, userAuthorities...)
-
-		hostAuthorities, err := client.GetCertAuthorities(services.HostCA, rc.withSecrets)
-		if err != nil {
-			return nil, trace.Wrap(err)
+		return &reverseTunnelCollection{tunnels: []services.ReverseTunnel{tunnel}}, nil
+	case services.KindCertAuthority:
+		if rc.ref.SubKind == "" && rc.ref.Name == "" {
+			var allAuthorities []services.CertAuthority
+			for _, caType := range types.CertAuthTypes {
+				authorities, err := client.GetCertAuthorities(caType, rc.withSecrets)
+				if err != nil {
+					return nil, trace.Wrap(err)
+				}
+				allAuthorities = append(allAuthorities, authorities...)
+			}
+			return &authorityCollection{cas: allAuthorities}, nil
 		}
-		authorities = append(authorities, hostAuthorities...)
-
-		jwtSigners, err := client.GetCertAuthorities(services.JWTSigner, rc.withSecrets)
+		id := types.CertAuthID{Type: types.CertAuthType(rc.ref.SubKind), DomainName: rc.ref.Name}
+		authority, err := client.GetCertAuthority(id, rc.withSecrets)
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
-		authorities = append(authorities, jwtSigners...)
-
-		return &authorityCollection{cas: authorities}, nil
+		return &authorityCollection{cas: []services.CertAuthority{authority}}, nil
 	case services.KindNode:
 		nodes, err := client.GetNodes(rc.namespace)
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
-		return &serverCollection{servers: nodes}, nil
+		if rc.ref.Name == "" {
+			return &serverCollection{servers: nodes}, nil
+		}
+		for _, node := range nodes {
+			if node.GetName() == rc.ref.Name {
```

Comment:
just a convenience thing:
```suggestion
			if node.GetName() == rc.ref.Name || node.GetHostname() == rc.ref.Name {
```

Reply:
I'll add this to each of the `services.Server` lookups.